inbm/dispatcher-agent/dispatcher/sota/snapshot.py

```diff
@@ -223,10 +225,13 @@ def recover(
         dispatcher_state.clear_dispatcher_state()
         if self.snap_num:
             self._rollback_and_delete_snap()
+            logger.debug("Rebooting to recover from failed SOTA...")
+            rebooter.reboot()
```

Comment:
**Suggestion:** The recover method in DebianBasedSnapshot class logs the message "Rebooting to recover from failed SOTA..." before actually checking if a reboot is necessary. This message should be logged only when the reboot condition is true to avoid confusion in the logs. [enhancement]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
            if self.snap_num:
                self._rollback_and_delete_snap()
                if rebooter.is_reboot(self._device_reboot):    
                    logger.debug("Rebooting to recover from failed SOTA...")
                    rebooter.reboot()
```